usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ValidateAllOptionalFileName.java

```diff
@@ -0,0 +1,40 @@
+package org.mobilitydata.gtfsvalidator.usecase;
+
+import org.mobilitydata.gtfsvalidator.usecase.notice.OptionalFileFoundNotice;
+import org.mobilitydata.gtfsvalidator.usecase.port.GtfsSpecRepository;
+import org.mobilitydata.gtfsvalidator.usecase.port.RawFileRepository;
+import org.mobilitydata.gtfsvalidator.usecase.port.ValidationResultRepository;
+
+import java.util.List;
+
+public class ValidateAllOptionalFileName {
+
+    private final GtfsSpecRepository specRepo;
+    private final RawFileRepository rawFileRepo;
+    private final ValidationResultRepository resultRepo;
+    private final List<String> filenameList;
+
+    public ValidateAllOptionalFileName(final GtfsSpecRepository specRepo, final RawFileRepository rawFileRepo,
+                                       final ValidationResultRepository resultRepo, List<String> filenameList) {
+        this.specRepo = specRepo;
+        this.rawFileRepo = rawFileRepo;
+        this.resultRepo = resultRepo;
+        this.filenameList = filenameList;
+    }
+
+    public List<String> execute() {
+
+        specRepo.getOptionalFilenameList().stream()
+                .filter(optionalFilename -> rawFileRepo.getFilenameAll().contains(optionalFilename))
+                .forEach(optionalFilename -> {
+                            filenameList.add(optionalFilename);
+                            resultRepo.addNotice(new OptionalFileFoundNotice(optionalFilename,
+                                    "I002",
```

Comment:
This is a separate issue, but I'd refactor `I002` and all other INFO, WARNINGS, ERRORS to variables ASAP. Otherwise we're just creating more work for ourselves every time we add a new one as a string. If we don't have an issue for this yet (I didn't see one scanning the list), could you create one?

Reply:
Ok assuming you meant refactoring Ids, that is already the case as can be seen here for errors
https://github.com/MobilityData/gtfs-validator/blob/aa2634e82478cf7ce88e79018d0eb787f923b782/usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/notice/base/ErrorNotice.java#L23
This specific Info Id was an overlook

Reply:
Ah, yes, good!